### ISSUE TITLE
no matter the XML file Error in rbind(deparse.level, ...) 

### ISSUE THREAD

**Issue by verajosemanuel** [jsugarelli/flatxml#1]
any xml file I've tried to read using:

`flatxml::fxml_importXMLFlat(filepath)`

I get an error:

```
Error in rbind(deparse.level, ...) : 
  numbers of columns of arguments do not match
```

sometimes with very simple files as this one:

```
<?xml version='1.0' encoding='UTF-8'?>
<read>	
<control>		
<dnp>1</dnp>		
<cons>2</cons>		
<cul>0</cul>
</control>
</read>
```

**Comment by jsugarelli** [comment]
Hi Jose,

Unfortunately I cannot reproduce your error. I took your example, put it into an XML file (`test.xml`) and then did

`z <- fxml_importXMLFlat("test.xml")`

This results in `z` being
`    elem. elemid. attr. value. level1  level2 level3`
`1    read       1    NA   <NA>   read    <NA>   <NA>`
`2 control       2    NA   <NA>   read control   <NA>`
`3     dnp       3    NA      1   read control    dnp`
`4    cons       4    NA      2   read control   cons`
`5     cul       5    NA      0   read control    cul`

which is correct, as far as I can see.

Can you provide another example?

Best regards

Joachim

**Comment by verajosemanuel** [comment]
here, an usual XML file obtained from public open data cartographical services in my country.

```
<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns:X="http://www.opengis.net/kml/2.2">
  <Document>
    <name>0240304VK1804S</name>
    <Style id="sn_ICONO_OVC">
      <IconStyle>
        <color>ffb2c0ff</color>
        <scale>1.2</scale>
        <Icon>
          <href>http://ovc.catastro.meh.es/Cartografia/WMS/ICONO_OVC.png</href>
        </Icon>
      </IconStyle>
      <LabelStyle>
        <scale>0.7</scale>
      </LabelStyle>
    </Style>
    <Style id="sh_ICONO_OVC">
      <IconStyle>
        <color>ffb2c0ff</color>
        <scale>1.4</scale>
        <Icon>
          <href>http://ovc.catastro.meh.es/Cartografia/WMS/ICONO_OVC.png</href>
        </Icon>
      </IconStyle>
      <LabelStyle>
        <scale>0.8</scale>
      </LabelStyle>
    </Style>
    <StyleMap id="msn_ICONO_OVC">
      <Pair>
        <key>normal</key>
        <styleUrl>#sn_ICONO_OVC</styleUrl>
      </Pair>
      <Pair>
        <key>highlight</key>
        <styleUrl>#sh_ICONO_OVC</styleUrl>
      </Pair>
    </StyleMap>
    <Visible>1</Visible>
    <LookAt>
      <longitude>-4.05972570914544</longitude>
      <latitude>40.5007103507443</latitude>
      <range>54.7199999999721</range>
      <tilt>0</tilt>
      <heading>0</heading>
    </LookAt>
    <Placemark>
      <name>CL SOROLLA, 7  VALDEMORILLO (MADRID)</name>
      <Snippet>0240304VK1804S</Snippet>
      <description><![CDATA[<font size=+1>Consultar en la Sede Electrónica<br>del Catastro la parcela:</font><br><font size=+2><A href="https://www1.sedecatastro.gob.es/CYCBienInmueble/OVCListaBienes.aspx?del=28&muni=160&rc1=0240304&rc2=VK1804S">0240304VK1804S</a></font>]]></description>
      <styleUrl>#msn_ICONO_OVC</styleUrl>
      <Point>
        <extrude>1</extrude>
        <altitudeMode>relativeToGround</altitudeMode>
        <coordinates>-4.05972570914544,40.5007103507443,1</coordinates>
      </Point>
    </Placemark>
  </Document>
</kml>


```

Related to the previous test file, I get the results fine on second try, but converting it to data frame with

 `flatxml::fxml_toDataFrame(z,siblings.of = 3)
`

gave me:

`data frame with 0 columns and 0 rows`

**Comment by jsugarelli** [comment]
Jose,

Thanks. I found the issue(s) and fixed it. I uploaded a version 0.0.2 to CRAN and will keep you posted when its live.

By the way, you would need to call `flatxml::fxml_toDataFrame(z,siblings.of = 2)` as the data records are on level 2 of your file, only the fields (represented by XML elements) are on level 3.

**Comment by verajosemanuel** [comment]
Hi

Thanks for your quick reponse fixing the bug. It's a great package.

Regarding the call to sibling levels no matter the number I always get empty dataframe:

```
> flatxml::fxml_toDataFrame(z,siblings.of =2)
data frame with 0 columns and 0 rows
> flatxml::fxml_toDataFrame(z,siblings.of =1)
data frame with 0 columns and 0 rows
```

regards

**Comment by jsugarelli** [comment]
Hi,

yes, the problem was the following: if there is only one data row, that data row does not make it into the dataframe.

This is also fixed with version 0.0.2.

Best regards

Joachim

**Comment by jsugarelli** [comment]
Version 0.0.2 is now on CRAN. Thanks for your input!
